fix(css): update container query pages (#35767)
mdn/content: files/en-us/web/css/css_containment/container_size_and_style_queries/index.md

## Naming containers

> A `<container-condition>` can include an optional case-sensitive {{cssxref("container-name")}}. A container name makes the container condition more specific. Such container condition is evaluated against only the elements that have `container-name` set to the name.

Reviewer: ```suggestion
A `<container-condition>` can include an optional case-sensitive {{cssxref("container-name")}}. A container name makes the container condition more specific — it is evaluated only against elements that have that name set in the `container-name` property.
```

The {{cssxref("container-name")}} property specifies a list of query container names that can be used in `@container` rules. The `<container-name>` is a case-sensitive {{cssxref("ident")}}. The container names enable targeting any container ancestor of the element. Without a container name the query matches only the nearest container ancestor.

```css

> @container [ <container-name> ]? <container-query> {

Reviewer: This formal syntax isn't quite right. Check out the latest at https://drafts.csswg.org/css-conditional-5/#container-rule

Author: Yes, the expanded version of 
`@container <container-condition>#` is 
`@container [[ <container-name> ]? <container-query>]#`.

  /* <stylesheet> */
}
```

After you add names to your `@container` at rules, you can use the {{cssxref("container-name")}} property or the {{cssxref("container")}} shorthand to target specific container elements. Styles inside the named `@container` at rules will be applied to only the matching elements inside the containers with the same names that satisfy the container query.

```css
@container card (orientation: landscape) {
  /* styles */
}

.todo-panel > li {
  container-type: inline-size;
  container-name: card;
}
```

> In the above example, the styles within the container query block will apply to the descendants of all {{htmlelement("li")}} elements with a width that is greater than their height. Note that if there are other elements, such as `.bottom-panel > .card`, with `container-name: card` applied to them that match the size query, the styles will also be applied to those elements' descendants.

Reviewer: ```suggestion
In the above example, the styles within the container query block will apply to the descendants of all {{htmlelement("li")}} elements with a width that is greater than their height. Note that other elements with `container-name: card` applied to them that match the size query will also have these styles applied to their elements' descendants.
```

I didn't think the spec example was very helpful — it made the sentence harder to read and also made me think "oh, is that example in one of the code blocks somewhere near"?

}
```

In the above example, the element has two container names, `wide` and `narrow`. The descendants of any elements with `class="sizeContainer"` will get the styles from the `wide` or `narrow` query applied (or both if an element is precisely 20em wide).

> The default value `container-type: normal` prevents the container from being a query container, meaning the element is not a size container but it can still be a [style container](#container_style_queries). The default value `container-name: none` states the container has no name, but it does not prevent the element from matching unnamed queries.

Reviewer 2: > The default value `container-type: normal` prevents the container from being a query container

@OnkarRuikar, this is wrong/confusing. It only prevents the container from being queried with a size query, but it can still be a "query container" for style queries.

Author: We mentioned `but it can still be a style container` in the same sentence.

More explicit wording could be:
> The default value `container-type: normal` prevents the container from being a query container for size queries, meaning the element is not a size container. However, the container remains a [style container](#container_style_queries) that is a query container for style queries.

Could you [submit a PR](https://github.com/mdn/content/edit/main/files/en-us/web/css/css_containment/container_size_and_style_queries/index.md)?

Reviewer 2: Done: #35871. Thank you!
